### DISCUSSION TITLE
How can I get the current URL or Function when i use a ? sugar

### DISCUSSION THREAD

**Discussion by BarneyKY** [actix/actix-web#3426] [Q&A]
I wrote my own error struct to log an error like this:

```rust
#[derive(Debug)]
struct MyError {}

impl fmt::Display for MyError {
    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
        write!(f, "{}", self)
    }
}

impl From<std::io::Error> for MyError {
    fn from(err: std::io::Error) -> Self {
        log::error!("{}", err);
        Self {}
    }
}

impl error::ResponseError for MyError {
    fn status_code(&self) -> StatusCode {
        StatusCode::OK
    }

    fn error_response(&self) -> HttpResponse {
        HttpResponse::Ok().body("This is an error response")
    }
}
```

```rust
#[get("/")]
async fn manual() -> Result<HttpResponse, MyError > {
    fs::read_dir("/asfafawffaw")?;
    Ok(HttpResponse::Ok().body("Hello world!"))
}
```

Now the error occurs and i log it ,but how can i  find out where it is,can i get the URL or the real line and file name ?

Thanks!

**Comment by robjtede** [comment]
It's not possible using only `?`. Perhaps, a simple constructor for the error type that takes the source error `E` and a `&HttpRequest` would allow you to keep boilerplate to a minimum. That way you can do `.map_err(|err| MyError::new(err, &req))?`.

If you want to also collect file/line number info you're gonna start wanting a macro_rules macro, I think.

**Reply by BarneyKY** [reply]
I have already used a macro_rules macro, this code is too redundant, repeating over and over again.
All i want is just making it simple,but seems it is impossible.

Thanks for your reply!

**Reply by robjtede** [reply]
It's a language limitation, unfortunately.
